From f39c5f12a696e30621e7351d6a014c4f5d621233 Mon Sep 17 00:00:00 2001
From: Daniel Wang <wonderfly@google.com>
Date: Mon, 13 Mar 2017 15:21:58 -0700
Subject: [PATCH] CHROMIUM: v4l: Add config store support

This patch adds pre v4.19 H264 header as legacy where new h264 headers will
be added in a different patch for Chromium to support both versions later.
This patch also adds config store support.

Change-Id: I5968831097ab093391783565c821266c4a337f00
Signed-off-by: Francois Buergisser <fbuergisser@chromium.org>
[wenst@chromium.org: Drop H264 changes and only keep config store for
 Rockchip libv4lplugins]
Signed-off-by: Chen-Yu Tsai <wenst@chromium.org>
---
 include/uapi/linux/videodev2.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/uapi/linux/videodev2.h b/include/uapi/linux/videodev2.h
index 16a0c343842c..f5cc6e7abc10 100644
--- a/include/uapi/linux/videodev2.h
+++ b/include/uapi/linux/videodev2.h
@@ -905,6 +905,7 @@ struct v4l2_plane {
  * @length:	size in bytes of the buffer (NOT its payload) for single-plane
  *		buffers (when type != *_MPLANE); number of elements in the
  *		planes array for multi-plane buffers
+ * @config_store: this buffer should use this configuration store
  * @request_fd: fd of the request that this buffer should use
  *
  * Contains data exchanged by application and driver using one of the Streaming
@@ -929,7 +930,7 @@ struct v4l2_buffer {
 		__s32		fd;
 	} m;
 	__u32			length;
-	__u32			reserved2;
+	__u32			config_store;
 	union {
 		__s32		request_fd;
 		__u32		reserved;
@@ -1590,6 +1591,7 @@ struct v4l2_ext_controls {
 	union {
 #ifndef __KERNEL__
 		__u32 ctrl_class;
+		__u32 config_store;
 #endif
 		__u32 which;
 	};
@@ -1632,6 +1634,8 @@ enum v4l2_ctrl_type {
	V4L2_CTRL_TYPE_U8	     = 0x0100,
	V4L2_CTRL_TYPE_U16	     = 0x0101,
	V4L2_CTRL_TYPE_U32	     = 0x0102,
+
+	V4L2_CTRL_TYPE_PRIVATE	     = 0xffff,
 };
 
 /*  Used in the VIDIOC_QUERYCTRL ioctl for querying controls */
-- 
2.39.2

